packages/@aws-cdk/aws-events-targets/lib/sqs.ts

```diff
@@ -0,0 +1,47 @@
+import events = require('@aws-cdk/aws-events');
+import iam = require('@aws-cdk/aws-iam');
+import sqs = require('@aws-cdk/aws-sqs');
+
+/**
+ * Customize the SQS Queue Event Target
+ */
+export interface SqsQueueProps {
+  /**
+   * The message to send to the queue
+   *
+   * @default the entire CloudWatch event
+   */
+  readonly message?: events.RuleTargetInput;
+}
+
+/**
+ * Use an SQS Queue as a target for AWS CloudWatch event rules.
+ *
+ * @example
+ *
+ *    // publish to an SQS queue every time code is committed
+ *    // to a CodeCommit repository
+ *    repository.onCommit(new targets.SqsQueue(queue));
+ *
+ */
+export class SqsQueue implements events.IRuleTarget {
+  constructor(public readonly queue: sqs.IQueue, private readonly props: SqsQueueProps = {}) {
+  }
+
+  /**
+   * Returns a RuleTarget that can be used to trigger this SQS queue as a
+   * result from a CloudWatch event.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/resource-based-policies-cwe.html#sqs-permissions
+   */
+  public bind(_rule: events.IRule): events.RuleTargetProperties {
+    // deduplicated automatically
+    this.queue.grantSendMessages(new iam.ServicePrincipal('events.amazonaws.com'));
```

Comment:
We also need the `arnEquals` condition? 
This basically allows any CWE rule to send messages to the queue... I think we can do better, no?

Reply:
I'm working on it, but I'm encountering an error in generating the resource...just a question: SNS does not have the same *problem*? 🤔

Reply:
What's the problem you are seeing?

Yes, could be that we have the same issue (and we should fix!)

Reply:
> What's the problem you are seeing?

Ok, I'm trying to implement the least privilege principle as discussed above. To do it, we have to put a condition like this in the policy of the SQS queue, if I'm not wrong

```
"Condition": {
    "ArnEqual": {
        "SourceArn": {
            "Fn::GetAtt": [
                "MyRuleA44AB831",
                "Arn"
            ]
        }
    }
},
```

Thus, the binding method has to access the `ruleArn` but I don't know why it's something I cannot do. I tried to access directly to the property of the `_rule` by looking for `ruleArn` property, I can correctly build everything but during test I get the following error (even if I can see a partial synth of the template).

```
81 |   public assertOrThrow(inspector: StackInspector) {
82 |     if (!this.assertUsing(inspector)) {
83 |       throw new Error(this.generateErrorMessage());
   |             ^
84 |     }
85 |   }
```

Then I thought to go the other way around by creating a `singletonEventRoleWithConditions` (or optional parameter): maybe this second option doesn't make sense since we have to limit the access from the `events.amazon.com` service principal - but conditioned to the specific rule we are creating. Maybe the problem I'm encountering is due to the lifecycle of the rule that still *doesn't exist* when I'm binding the target. 

In the meanwhile, I also ported the queue as a `props`, but it didn't bring me so far.

But.... it could be that I misunderstood everything and the condition needs to Reference to the queue instead: but in this way, we would limit the *actor* that will act on the resource instead of protecting the *resource* to be accessed only by a specific agent, right?

So... I'm confused, maybe lost in a glass of water 😅

> Yes, could be that we have the same issue (and we should fix!)

SNS is not the only one: I think that also lambda, codebuild, codepipeline. In the cluster the it's differente because of the chain of executionRole of the `taskDefinition` 🤔

Reply:
If you merge from master, you should be able to make a `ServicePrincipal` with the condition you need:

```ts
this.queue.grantSendMessages(new iam.ServicePrincipal('events.amazonaws.com', {
  conditions: {
    ArnEqual: { "aws:SourceArn": rule.ruleArn } 
  }
});
```

Whether it's `aws:SourceArn` or `SourceArn`, I'm a little fuzzy on atm.

> during test I get the following error (even if I can see a partial synth of the template).

You quoted the code that throws the error, but not the error message, so it's hard to say what's going on there :)